### ISSUE TITLE
Support running doc tests serially

### ISSUE THREAD

**Issue by llamicron** [palfrey/serial_test#30]
This crate is very useful for my use-case (interacting with hardware), but I have quite a few doc tests that can't be run unless serially. Does this crate support running doc tests serially? If not, could that be supported?

**Comment by palfrey** [comment]
I've just been experimenting with this, and the following seems to work ok (for values of "doesn't seem to fail"). OTOH, I've been trying a few experiments, and so far I *think* rustdoc tests are maybe run separately enough that the `serial_test` usage of a common lock structure just doesn't actually work. Not sure how to fix this barring digging into the rustdoc internals and trying to figure out how to share data between rustdoc tests.

```
//! ```
//! # #[macro_use] extern crate serial_test;
//! # #[serial_test::serial]
//! # fn main() {
//! assert!(true)
//! # }
//! ```
```

I've marked the boilerplate bits (everything other than the `assert!` in this minimal check) as "don't display".

**Comment by palfrey** [comment]
Yup, I've just confirmed this. Between https://quietmisdreavus.net/code/2018/02/23/how-the-doctests-get-made/ and https://github.com/rust-lang/rust/blob/c4e33b51c1a2d5e599b949fa3006467b88df253a/src/librustdoc/doctest.rs#L233 I've managed to confirm that `rustdoc` basically makes separate executables for each doctest, and so the standard serial_test methods of locking won't work.

OTOH, I could just use something like https://docs.rs/fslock and make a new attribute that lets you do file-based locking, but that's a notable change that would take me a bit to do and I've got a few other things on my plate at the moment. I might get around to doing this eventually, but if you felt like contributing patches for that it would be appreciated :)
